app/src/main/java/net/squanchy/onboarding/location/LocationOnboardingActivity.java

```diff
@@ -85,69 +85,80 @@ private void disableUi() {
 
     private void onGoogleConnectionFailed() {
         Timber.e("Google Client connection failed");
-        Snackbar.make(contentRoot, R.string.onboarding_error_google_client_connection, Snackbar.LENGTH_LONG).show();
+        Snackbar.make(contentRoot, R.string.proximity_error_google_client_connection, Snackbar.LENGTH_LONG).show();
     }
 
     private ProximityPreconditions.Callback proximityPreconditionsCallback() {
         return new ProximityPreconditions.Callback() {
 
             @Override
-            public void allChecksPassed() {
-                startRadarAndFinish();
+            public void notOptedIn() {
+                showFatalProximityError(R.string.proximity_error_not_opted_in);
+            }
+
+            @Override
+            public void featureDisabled() {
```

Comment:
can we have `Timber.e` something here? This shouldn't happen I suppose (as in you shouldn't get here to start with if the feature is disabled)